history-network.md

```diff
@@ -60,31 +60,37 @@ Each of these data items are represented as a key-value pair. Denote the key for
 
 All `content` items are transmitted as RLP-encoded byte arrays.
 
-We derive a `content-id` from the `content-key` as `H(content-key)` where `H` denotes the SHA-256 hash function, which outputs 32-byte values. The `content-id` represents the key in the DHT that we use for `distance` calculations.
-
-All `content-key` values are encoded and decoded according to SSZ sedes.
+All `content-key` values are encoded and decoded as an [`SSZ Union`](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#union) type.
+```
+content-key = Union[blockheader, blockbody, receipt]
+serialized-content-key = serialize(content-key)
+```
 
 #### Block Header
 
 ```
-content-key  = Container(chain-id: uint16, content-type: uint8, block-hash: Bytes32)
-content-type = 0x01
+selector = 0x00
+content-key  = Container(chain-id: uint16, block-hash: Bytes32)
 ```
 
 #### Block Body
 
 ```
-content-key  = Container(chain-id: uint16, content-type: uint8, block-hash: Bytes32)
-content-type = 0x02
+selector = 0x01
+content-key  = Container(chain-id: uint16, block-hash: Bytes32)
 ```
 
 #### Receipts
 
 ```
-content-key  = Container(chain-id: uint16, content-type: uint8, block-hash: Bytes32)
-content-type = 0x03
+selector = 0x02
+content-key  = Container(chain-id: uint16, block-hash: Bytes32)
 ```
 
+#### Content ID
+
+We derive a `content-id` from the `content-key` as `H(serialized-content-key)` where `H` denotes the SHA-256 hash function, which outputs 32-byte values. The `content-id` represents the key in the DHT that we use for `distance` calculations.
+
```

Comment:
Why is it necessary to take the hash of `serialized-content-key` vs just `content-key`, as previously specified? Currently I have the implementation from key->id as just taking the hash of the concatenation of each field of the (unserialized) key, ie `H(chain-id | block-hash)`